vite.config.ts

```diff
@@ -0,0 +1,22 @@
+import { resolve } from 'path'
+import { defineConfig } from 'vite'
+import typescript from 'rollup-plugin-typescript2'
```

Comment:
[typescript2](https://www.npmjs.com/package/rollup-plugin-typescript2) is a rewrite of [@rollup/typescript](https://www.npmjs.com/package/@rollup/plugin-typescript)

> Rollup plugin for typescript with compiler errors.
> This is a rewrite of original rollup-plugin-typescript, starting and borrowing from [this fork](https://github.com/alexlur/rollup-plugin-typescript).
> This version is somewhat slower than original, but it will print out typescript syntactic and semantic diagnostic messages (the main reason for using typescript after all).

I chose it since it seems to be being used more than the original